js/components/search.js

```diff
@@ -5,85 +5,189 @@ $( document ).ready(function() {
   //
   // By Sean Herron <sean@herron.io>
   //
-  function GetURLParameter(sParam) {
-    var sPageURL = window.location.search.substring(1);
-    var sURLVariables = sPageURL.split('&');
-    for (var i = 0; i < sURLVariables.length; i++) {
-      var sParameterName = sURLVariables[i].split('=');
-      if (sParameterName[0] == sParam) {
-        return sParameterName[1];
-      }
-    }
+
+  /**
+   * Returns parameter from GET query string
+   *
+   * Note that in application/x-www-form-urlencoded strings
+   * (such as the GET query), a `+` will replace `%20` so we
+   * need to _untranslate_ that back into `%20` before decoding.
+   * The _actual_ `+` is safely encoded as `%2B`, however, so
+   * we do not have to worry about losing `+`s
+   *
+   * @param {string} param - Which value to extract by name in name=value
+   * @returns {string} Value of first requested param else ''
+   */
+  function getQueryParam(param) {
+    return decodeURIComponent(window.location.search
+      .slice(1) // drop the leading '?'
+      .split('&') // split into name=value pairs
+      .map(function(pair) { return pair.split('=') }) // split pairs into [name, value]
+      .filter(function(pair) { return pair[0] === param }) // return matching pairs
+      .map(function(pair) { return pair[1] }) // but only return the value
+      .concat( '' ) // add a default in case didn't find any
+      .shift() // return first match or empty string if none available
+      .replace(/[+]/g, '%20') // untranslate `+` into `%20` (see above)
+    );
   }
 
-  var query = GetURLParameter('q') || '',
-    apiKEY = eiti.beckleyApiKey;
-
-  $('.search-string').append(query.replace(/%20/g,' '));
-  $('.site-search-text').attr('value',query.replace(/%20/g,' '));
-  if (query) {
```

Comment:
This line right here is why it appears like I changed the whole file. Instead of wrapping everything inside a conditional, I checked for the non-existence of the query and early-aborted, leaving everything else un-nested.